crates/orchestrator/src/tests/jobs/state_update_job/mod.rs

```diff
@@ -21,16 +22,98 @@ use crate::jobs::{
 use crate::config::{config, config_force_init};
 use crate::constants::{BLOB_DATA_FILE_NAME, SNOS_OUTPUT_FILE_NAME};
 use crate::data_storage::MockDataStorage;
-use crate::jobs::state_update_job::utils::hex_string_to_u8_vec;
+use crate::jobs::constants::JOB_METADATA_STATE_UPDATE_LAST_FAILED_BLOCK_NO;
+use crate::jobs::state_update_job::utils::{fetch_blob_data_for_block, hex_string_to_u8_vec};
+use crate::tests::common::{default_job_item, get_storage_client};
+use crate::tests::config::TestConfigBuilder;
 use httpmock::prelude::*;
 use lazy_static::lazy_static;
+use utils::env_utils::get_env_var_or_panic;
 
 lazy_static! {
     pub static ref CURRENT_PATH: PathBuf = std::env::current_dir().unwrap();
 }
 
 pub const X_0_FILE_NAME: &str = "x_0.txt";
 
+// ================= Exhaustive tests (with minimum mock) =================
+
+#[rstest]
+#[should_panic(expected = "Could not find current attempt number.")]
```

Comment:
Should panic? Why?

Reply:
Because the function should panic when attempt number is not found in job metadata.

Reply:
Wouldn't returning an error be better?

Reply:
yeah so we have a PR that is pending and includes the implementation of thiserror library. So in that case I will update the test to return that particular error. So for now this is added as panic!